ui/pages/confirmations/components/confirm/scroll-to-bottom/scroll-to-bottom.tsx

```diff
@@ -31,16 +33,35 @@ const ScrollToBottom = ({ children }: ContentProps) => {
   const t = useContext(I18nContext);
   const dispatch = useDispatch();
   const currentConfirmation = useSelector(currentConfirmationSelector);
+  const previousId = usePrevious(currentConfirmation?.id);
 
   const {
     hasScrolledToBottom,
     isScrollable,
     isScrolledToBottom,
     onScroll,
     scrollToBottom,
+    setHasScrolledToBottom,
     ref,
   } = useScrollRequired([currentConfirmation?.id]);
 
+  /**
+   * Scroll to the top of the page when the confirmation changes. This happens
+   * when we navigate through different confirmations. Also, resets hasScrolledToBottom
+   */
+  useEffect(() => {
+    if (isEqual(previousId, currentConfirmation?.id)) {
+      return;
+    }
+
+    setHasScrolledToBottom(false);
+
+    const scrollTo = (ref?.current as null | HTMLDivElement)?.scrollTo;
+    if (typeof scrollTo === 'function') {
+      (ref?.current as null | HTMLDivElement)?.scrollTo(0, 0);
+    }
+  }, [currentConfirmation?.id]);
```

Comment:
Are we missing dependencies for `setHasScrolledToBottom` and `ref` as they come from another hook?

Reply:
ooh nice catch. Thought this would be caught by our linter. updated https://github.com/MetaMask/metamask-extension/pull/24435/commits/60ff8173d5793f8a6a509d31f3edb2408e81005f